pkg/pillar/devicenetwork/dhcpcd.go

```diff
@@ -165,11 +165,22 @@ func doDhcpClientActivate(nuc types.NetworkPortConfig) bool {
 			log.Errorf("doDhcpClientActivate: request failed for %s\n",
 				nuc.IfName)
 		}
+		// Wait for a bit then give up
+		waitCount := 0
+		failed := false
 		for !dhcpcdExists(nuc.IfName) {
 			log.Warnf("dhcpcd %s not yet running", nuc.IfName)
+			waitCount++
+			if waitCount >= 3 {
+				log.Errorf("dhcpcd %s not yet running", nuc.IfName)
+				failed = true
+				break
+			}
 			time.Sleep(10 * time.Second)
```

Comment:
Do you want to sleep for less so that the loop will complete in less than 30 seconds? We try to make sure all operations finish in 30 seconds or so. sleep(2) might be sufficient??

Reply:
@eriknordmark 30 seconds is what DH_CLIENT case also uses. My thinking was that there should have been a reason why 30 seconds is being used for DH_CLIENT case, that is the reason I opted to use the same wait time for DH_STATIC also. May be I can reduce the sleep to 15 seconds ? 5 seconds in each iteration?

Reply:
Aren't you also waiting in the code which calls this dhcp routine? If you do, then what is the total time you can wait?
Note that what you have is better than now, since currently it waits forever.

Reply:
@eriknordmark While testing new DPC we have the PendTimer (default 30 secs) that fires VerifyPending. But as part of this we check if there are any usable IP addresses and not for existence of dhcpcd for the port of interest. Adding both I think in the worst case we wait for 60 seconds in a DPC with one port. But the PendTimer (30 secs) in non-blocking. The waiting that we do as part do doDhcpClientActivate (30 secs) is blocking though.